kie-server-parent/kie-server-services/kie-server-services-drools/src/main/java/org/kie/server/services/drools/DroolsKieSessionLookupHandler.java

```diff
@@ -42,32 +44,36 @@ public CommandExecutor lookupKieSession(String kieSessionId, KieContainerInstanc
                         break;
                     case STATELESS:
                         ks = ((KieContainerImpl)containerInstance.getKieContainer()).getStatelessKieSession(kieSessionId);
-
                         break;
                 }
             }
+            return ks;
+        }
+        return null;
+    }
 
-            PrometheusKieServerExtension extension = (PrometheusKieServerExtension)registry.getServerExtension(PrometheusKieServerExtension.EXTENSION_NAME);
-            if (extension != null && ks != null) {
-                RuleRuntimeEventManager eventManager = (RuleRuntimeEventManager) ks;
+    @Override
+    public void postLookupKieSession(String kieSessionId, KieContainerInstance containerInstance, CommandExecutor ks, KieServerRegistry registry) {
```

Comment:
I moved the logic to add PrometheusMetricsDroolsListener to ksession from lookupKieSession() to postLookupKieSession() because when a client doesn't specify "ksession name", lookupKieSession() is not executed and the default ksession is found by KieServerUtils.getDefaultKieSession(). So we make sure that PrometheusMetricsDroolsListener is added in both cases.